plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -482,11 +483,21 @@ public long deleteDocuments(RemoteTableName remoteTableName, TupleDomain<ColumnH
 
     public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoColumnHandle> columns)
     {
+        Set<MongoColumnHandle> projectedColumns = tableHandle.getProjectedColumns();
+        checkArgument(projectedColumns.isEmpty() || projectedColumns.equals(new HashSet<>(columns)), "Columns and projected columns must be equal");
+
+        MongoCollection<Document> collection = getCollection(tableHandle.getRemoteTableName());
         Document output = new Document();
-        for (MongoColumnHandle column : columns) {
-            output.append(column.getName(), 1);
+        for (MongoColumnHandle column : getSufficientColumns(columns)) {
+            if (column.isDbRefField()) {
+                // DBRef dereference field can be projected using parent name or using $ref, $id, $db accessor.
+                // project it via qualified parent name
```

Comment:
"project it via qualified parent name because the connector can't distinguish between DBRef and ROW type"